src/modules/joystick/sdl/Joystick.cpp

```diff
@@ -509,7 +515,7 @@ bool Joystick::setVibration(float left, float right, float duration)
 	Uint32 length = SDL_HAPTIC_INFINITY;
 	if (duration >= 0.0f)
 	{
-		float maxduration = std::numeric_limits<Uint32>::max() / 1000.0f;
+		float maxduration = (float)std::numeric_limits<Uint32>::max() / 1000.0;
```

Comment:
Won't this still cast `2^32-1` to a float, and then do the division? That was the original issue with the code that you wanted to change, because floats don't have enough precision to represent 2^32-1 (it ends up as 2^32 instead).

I believe adding parentheses around it would resolve it because I think it will cast 2^32-1 to a double instead of a float, for the division:
```suggestion
		float maxduration = (float) (std::numeric_limits<Uint32>::max() / 1000.0);
```